src/components/user/unavailability/unavailability.service.ts

```diff
@@ -15,59 +22,97 @@ import {
 export class UnavailabilityService {
   constructor(
     @Logger('unavailability:service') private readonly logger: ILogger,
-    private readonly db: DatabaseService
+    private readonly db: DatabaseService,
+    private readonly config: ConfigService
   ) {}
 
   async create(
     { userId, ...input }: CreateUnavailability,
     session: ISession
   ): Promise<Unavailability> {
-    const id = generate();
-    const acls = {
-      canReadDescription: true,
-      canEditDescription: true,
-      canReadStart: true,
-      canEditStart: true,
-      canReadEnd: true,
-      canEditEnd: true,
-    };
+    const secureProps = [
+      {
+        key: 'description',
+        value: input.description,
+        addToAdminSg: true,
+        addToWriterSg: true,
+        addToReaderSg: true,
+        isPublic: false,
+        isOrgPublic: false,
+      },
+      {
+        key: 'start',
+        value: input.start,
+        addToAdminSg: true,
+        addToWriterSg: true,
+        addToReaderSg: true,
+        isPublic: false,
+        isOrgPublic: false,
+      },
+      {
+        key: 'end',
+        value: input.end,
+        addToAdminSg: true,
+        addToWriterSg: true,
+        addToReaderSg: true,
+        isPublic: false,
+        isOrgPublic: false,
+      },
+    ];
+
     try {
-      await this.db.createNode({
-        session,
-        type: Unavailability,
-        input: { id, ...input },
-        acls,
+      const createUnavailability = this.db
+        .query()
+        .call(matchRequestingUser, session)
+        .match([
+          node('root', 'User', {
+            active: true,
+            id: this.config.rootAdmin.id,
+          }),
+        ])
+        .call(
+          createBaseNode,
+          'Unavailability',
+          secureProps,
+          {
+            owningOrgId: session.owningOrgId,
+          },
+          [],
+          session.userId === this.config.rootAdmin.id
+        )
+        .return('node.id as id')
+        .asResult<{ id: string }>();
+
+      const createUnavailabilityResult = await createUnavailability.first();
```

Comment:
first() can return null if nothing was matched. I would add a null check to be safe. This also prevents the need to non-null assert below

Reply:
Oh I thought the entire call would just fail and get caught by the try...catch?  But I'm not 100%, I'll add the null check to be safe